src/core/manager.c

```diff
@@ -3568,14 +3570,14 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
                         else
                                 m->honor_device_enumeration = b;
 
-                } else if ((val = startswith(l, "show-status="))) {
+                } else if ((val = startswith(l, "show-status-overridden="))) {
```

Comment:
hmm, why did you rename the field? I mean, of course it has different semantics, but shouldn't we just leave the old name to do something somewhat reasonable during updates that to reexec? i.e. is there a reason behind renaming this serialization field besides "feels cleaner"?

Reply:
Because  "show-status"  always reflects the value set in the configuration file. Therefore this value is and should always be reloaded.

Serializing this field was actually a bug because, as explained in the commit message, because it locked the initial value read from the config file making impossible any later customization.

Reply:
I am not arguing which field should be serialized. I am argue under which name to do that. i.e. previously the value read from the config and the "overriden" value wasn't distuingished, now it is. When you runtime update your systemd from an old version to  anew version that includes your patch then this means we'd suddenly forget the overriden state, since we would not process the old `show-status` value at all, but that's where both the config and the overriden thing ended up in...

Hence I am proposing: read the show-status= serialization field into the show_status_overriden variable (instead of the show_status variable).

Reply:
What you propose would still be boggus.

There's no way to fix that properly because as you wrote the "overridden" state is lost after serializing.

So when deserializing we have only 2 options: either we assume that show-status value was overridden or it was not.

You suggest to assume that the value was overridden. That would still be problematic because if show-status value was actually not overridden and systemd was reexecuted into a new version then the old value would become suddenly the overridden value which would mean that any attempt to customize its value through the config file would not work. IOW the existing bug would not be fixed by updating systemd.

With the current patch, which assumes that the old value was not overridden, the previous use case would be fixed. However if show-status was actually overridden (via the signal API) after updating systemd the old value will still be restored but it will loose its "overridden" status. It means that the value could later be overridden by simply reloading the config.

Assuming that changing show-status via the signal API is pretty uncommon, especially since this API is pretty awkward to use, I would say that the first use case is more important to fix. Also the code is less confusing (or feel "cleaner" if you prefer).